gapis/api/transform2/control_flow.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/command_generator"
+	"github.com/google/gapid/gapis/config"
+)
+
+type ControlFlow struct {
+	transforms       []Transform
+	commandGenerator command_generator.CommandGenerator
+	tag              string
+	out              Writer
+}
+
+func NewControlFlow(tag string, commandGenerator command_generator.CommandGenerator, out Writer) *ControlFlow {
+	return &ControlFlow{
+		transforms:       make([]Transform, 0),
+		commandGenerator: commandGenerator,
+		out:              out,
+		tag:              tag,
+	}
+}
+
+func (cf *ControlFlow) AddTransform(transforms ...Transform) {
```

Comment:
It should not be possible to modify the transform chains inside a ControlFlowGenerator post construction. This opens a lot of questions about what to do when they get modified in the middle of operation. We can avoid addressing all those questions by saying they're passed once at construction time.